### PR TITLE
Relocate everything to make packaging work

### PR DESCRIPTION
**Pull request by webbnh**
This PR is an attempt to reconcile the existing problems with packaging, installation, and `PYTHONPATH`.

This PR takes the existing `relay` subdirectory, which contains the `__init__.py` file and therefore defines the `relay` Python package, and moves it into a `src` subdirectory.  (It also moves the existing `tests` directory which was a subdirectory of `relay` and makes it a peer of the new `src` directory at the top level; and, it moves the existing `relay/requirements.txt` file up a level to the `src` subdirectory.)  With the commensurate changes to the `setup.cfg` file, the packaging now includes the file-relay sources as well as the `relay` stub-executable, so that, when the package is installed, the `relay` package is copied into the `site-packages` directory as it should be and it is available without explicitly defining `PYTHONPATH`.  I'm not exactly sure what magic I've wrought here, but everything now seems to work as expected.
 
This PR also updates the `ci.yml` workflow to reflect the new directory structure (e.g., the locations of the respective `requirements.txt` files have changed, and the tests and sources are no longer in the same directory tree, so we have to request coverage for them separately) and to Pip-install the `relay` package from its sources (and to do it all in a single command).  This allows the unit tests to work without a `PYTHONPATH` definition, testing the package as any other client would.

### CODE REVIEW

File: tests/test_relay.py [hunk 1]
```diff
@@ -16,8 +16,8 @@
 import humanize
 import pytest
 
-# This file is in a subdirectory, so add the CUT's location to the path
-sys.path.append(str(Path(__file__).resolve().parent.parent.parent))
+# # This file is in a subdirectory, so add the CUT's location to the path
+# sys.path.append(str(Path(__file__).resolve().parent.parent.parent))
```
Comment (dbutenhof) [inline tests/test_relay.py:20]: I assume you'll remove those? And having commented them, flake8 believes you should have removed the import of `sys`...

**Review by dbutenhof — COMMENTED** [review]
Once you fix the "flake8y" code, I'll be interested in seeing whether the tests run. 😁

File: tests/test_relay.py (hunk 1 above)
Comment (webbnh) [reply tests/test_relay.py:20]: Um, yes.  (This PR should have been moved to draft mode when the tests didn't "just run" -- it's not quite ready for prime time, yet, although I am growing in understanding of packaging issues (and starting to develop _some_ sympathy (although, it **is** limited) for our friends on `pquisby`).)

The above was a hack driven by the original file layout.  With the new file layout, it is no longer necessary (the commenting-out was a temporary experiment, which mostly succeeded) and will be removed before this is merged.

**Comment by webbnh** [comment]
> Patch coverage: **`100.00%`** and project coverage change: **`-0.01%`** ⚠️

Such is the power of statistics:  by removing three previously covered lines, I increased the _fraction_ of non-covered lines, which therefore **decreased** the overall "coverage".  😆

In any case, this change is now ready for review.

File: tests/test_relay.py (hunk 1 above)
Comment (webbnh) [reply tests/test_relay.py]: All better now...although, sadly, this change [reduced](https://github.com/distributed-system-analysis/file-relay/pull/8#issuecomment-1699504354) the unit test coverage....  🤣

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**